core/src/transaction.rs

```diff
@@ -26,6 +26,51 @@ pub const TX_VERSION: Version = 0;
 // "NERVOSDAOINPUT0001" in hex mode.
 pub const ISSUING_DAO_HASH: H256 = h256!("0x4e4552564f5344414f494e50555430303031");
 
+pub struct CellKey([u8; 36]);
+
+impl CellKey {
+    pub fn calculate(tx_hash: &H256, index: u32) -> Self {
+        let mut key: [u8; 36] = [0; 36];
+        key[..32].copy_from_slice(tx_hash.as_bytes());
+        key[32..36].copy_from_slice(&index.to_le_bytes());
+        CellKey(key)
+    }
+
+    pub fn recover(&self) -> CellOutPoint {
+        Self::deconstruct(&self.0)
+    }
+
+    pub fn deconstruct(bytes: &[u8]) -> CellOutPoint {
```

Comment:
`to_cell_outpoint` is better?

Reply:
There are more than two methods returns `CellOutPoint`, which should be `to_cell_outpoint`?

- `recover(..)`: we have a `CellKey`, we recover the `CellOutPoint` from the `CellKey`

- `deconstruct(..)`: we don't have a `CellKey`, we can deconstruct / parse / deserialize a `CellOutPoint` from a slice input. `CellKey` is more like a tool or a utility struct.

- I wrote two methods, because we can avoid to create a temporary `[u8; N]` from a slice.